aspnet-core/src/AbpCompanyName.AbpProjectName.Application/Authorization/Accounts/AccountAppService.cs

```diff
@@ -43,12 +56,80 @@ public async Task<RegisterOutput> Register(RegisterInput input)
                 true // Assumed email address is always confirmed. Change this if you want to implement email confirmation.
             );
 
-            var isEmailConfirmationRequiredForLogin = await SettingManager.GetSettingValueAsync<bool>(AbpZeroSettingNames.UserManagement.IsEmailConfirmationRequiredForLogin);
+            var isEmailConfirmationRequiredForLogin = await SettingManager.GetSettingValueAsync<bool>(
+                AbpZeroSettingNames.UserManagement.IsEmailConfirmationRequiredForLogin);
 
             return new RegisterOutput
             {
                 CanLogin = user.IsActive && (user.IsEmailConfirmed || !isEmailConfirmationRequiredForLogin)
             };
         }
+
+        public async Task SendPasswordResetEmail(string userNameOrEmailAddress)
+        {
+            // validate if tenant is valid
+            var tenant = await _userRegistrationManager.GetActiveTenantAsync();
+
+            // use from absession since host is a possibility
+            var user = await _userManager.FindByNameOrEmailAsync(AbpSession.TenantId, userNameOrEmailAddress);
+            if (user == null)
+            {
+                // don't throw erros to avoid security issue
+                // pretend the email was sent
+                return;
```

Comment:
You can return abp user friendly exception if email was not found. Because user can make typo. And remove all unnecessary comments.

Reply:
@alirizaadiyahsi I honestly think it is a security flaw to inform if the e-mail exists or not. Nowadays it is common to inform that the user will receive an e-mail if it is registered on the website. If he made a typo, he can try again.